apps/finance/contracts/Finance.sol

```diff
@@ -717,25 +786,52 @@ contract Finance is EtherTokenConstant, IsContract, AragonApp {
         return _getRemainingBudget(_token) >= _amount && vault.balance(_token) >= _amount;
     }
 
+    function _currentPeriodId() internal view returns (uint64) {
+        // There is no way for this to overflow if protected by an initialization check
+        return periodsLength - 1;
+    }
+
     function _getRemainingBudget(address _token) internal view returns (uint256) {
         if (!settings.hasBudget[_token]) {
             return MAX_UINT;
         }
 
+        uint256 budget = settings.budgets[_token];
         uint256 spent = periods[_currentPeriodId()].tokenStatement[_token].expenses;
 
         // A budget decrease can cause the spent amount to be greater than period budget
         // If so, return 0 to not allow more spending during period
-        if (spent >= settings.budgets[_token]) {
+        if (spent >= budget) {
             return 0;
         }
 
-        return settings.budgets[_token].sub(spent);
+        // We're already protected from the overflow above
+        return budget - spent;
     }
 
-    function _currentPeriodId() internal view returns (uint64) {
-        // There is no way for this to overflow if protected by an initialization check
-        return periodsLength - 1;
+    function _nextPaymentTime(uint256 _paymentId) internal view returns (uint64) {
+        ScheduledPayment storage payment = scheduledPayments[_paymentId];
+
+        if (payment.executions >= payment.maxExecutions) {
+            return MAX_UINT64; // re-executes in some billions of years time... should not need to worry
+        }
+
+        // Split in multiple lines to circumvent linter warning
+        uint64 increase = payment.executions.mul(payment.interval);
+        uint64 nextPayment = payment.initialPaymentTime.add(increase);
+        return nextPayment;
+    }
+
+    // Syntax sugar
+
+    function _arr(address _a, address _b, uint256 _c, uint256 _d, uint256 _e, uint256 _f) internal pure returns (uint256[] r) {
+        r = new uint256[](6);
+        r[0] = uint256(_a);
+        r[1] = uint256(_b);
+        r[2] = _c;
+        r[3] = _d;
+        r[4] = _e;
+        r[5] = _f;
```

Comment:
Should we move this one to `ACLSyntaxSugar`?

Reply:
I had debated this, but this is such a super specific instance that I don't think it'd make sense to add back in `ACLSyntaxSugar`. Apps that have giant params lists like this will need their own syntax sugar :).